src/component/page/bruker-detaljer/veileder-panel/VeilederPanel.tsx

```diff
@@ -0,0 +1,98 @@
+import { EMDASH } from '../../../../utils/constants'
+import { isNotStartedOrPending, isRejected, isResolved, usePromise } from '../../../../utils/use-promise'
+import { IconLabel } from '../icon-label/IconLabel'
+import { TildelVeilederModal } from '../tildel-veileder-modal/TildelVeilederModal'
+import styles from './VeilederPanel.module.scss'
+
+interface Props {
+	deltaker: TiltakDeltakerDetaljer
+}
+
+export const VeilederPanel = ({ deltaker }: Props): React.ReactElement => {
+	const hentVeilederePromise = usePromise<AxiosResponse<Veileder[]>>(() => hentVeiledereForDeltaker(deltaker.id))
+	const [ veileder, setVeileder ] = useState<Veileder>()
+	const [ medveiledere, setMedveiledere ] = useState<Veileder[]>([])
+	const [ openModal, setOpenModal ] = useState(false)
+
+	useEffect(() => {
+		if (isResolved(hentVeilederePromise)) {
+			setVeileder(hentVeilederePromise.result.data.find(v => !v.erMedveileder))
+			setMedveiledere(hentVeilederePromise.result.data.filter(v => v.erMedveileder))
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ hentVeilederePromise.result ])
+
+	if (isNotStartedOrPending(hentVeilederePromise)) {
+		return <div className={styles.loader}><Loader size="2xlarge" /></div>
+	}
+
+	if (isRejected(hentVeilederePromise)) {
+		return <Alert variant="error">Kan ikke vise veiledere.</Alert>
+	}
+
+	const handleModalState = () => {
+		setOpenModal(prev => !prev)
+	}
+
+	const handleSubmit = (veiledere: Veileder[]) => {
+		setVeileder(veiledere.find(v => !v.erMedveileder))
+		setMedveiledere(veiledere.filter(v => v.erMedveileder))
+	}
+
+	return (
+		<Panel border className={styles.infoPanel}>
+
+			<Heading size="small" level="3" className={globalStyles.blokkXs}>Tiltaksarrangør</Heading>
+			<Heading size="xsmall" level="4" className={globalStyles.blokkXs}>Veileder</Heading>
+			<div className={cls(styles.contentBlock, globalStyles.blokkM)}>
+				<IconLabel
+					labelValue={veileder ? lagBrukerNavn(veileder.fornavn, veileder.mellomnavn, veileder.etternavn) : EMDASH}
+					icon={<People title="Veileder navn" />}
+					iconWrapperClassName={styles.iconWrapper}
+				/>
+			</div>
+
+			<Heading size="xsmall" level="4" className={globalStyles.blokkXs}>Medveiledere</Heading>
+			<div className={cls(styles.contentBlock, globalStyles.blokkM)}>
+				{medveiledere.length > 0 ? medveiledere.map(v => {
+					return <IconLabel
+						labelValue={lagBrukerNavn(v.fornavn, v.mellomnavn, v.etternavn)}
+						icon={<People title="Veileder navn" />}
+						iconWrapperClassName={styles.iconWrapper}
+						key={v.ansattId}
+					/>
+				})
+					: <IconLabel
+						labelValue={EMDASH}
+						icon={<People title="Veileder navn" />}
+						iconWrapperClassName={styles.iconWrapper}
+					/>
+				}
+			</div>
+			<Button variant="secondary" size="small" className={styles.knapp} onClick={handleModalState}>
+				<span className={styles.knappTekst}><AddPerson /> Endre</span>
```

Comment:
Her kan man evt vurdere å vente til promise er resolved før man tillater å trykke på knappen. Det vil være en teoretisk mulighet for at man trykker før man har fått svar og det vil kunne føre til rar tilstand hvor man kan begynne å gjøre endringer i modalen før man har sett hvem som var veileder før.

Reply:
Vet ikke helt om jeg forstår deg riktig. Akkurat nå så er det en spinner isteden for panelet hvis ikke promiset er resolved.

Reply:
Åja, nå skjønner jeg det. Ser spinneren nå så kommentaren min ble helt feil.. 

Var basert på samtalen over, hvor vi snakka om useState som ikke kan ha init verdi fordi veiledere er `undefined` først.
Det kan skje at `isNotStartedOrPending` og `isRejected` ikke slår til men dataene ikke er 
populert enda fordi når promise blir resolved så trigges useEffecten men det kan hende at man rekker å returnere før useEffecten er ferdig. 
Når jeg la til `if(!isResolved(hentVeilederePromise)) return <></>` over `return` i denne komponenten, så er dataene alltid populert i første render av `TildelVeilederModal`. 
Det går også å skrive 
`{veileder && <TildelVeilederModal
				deltaker={deltaker}
				veileder={veileder}
				medveiledere={medveiledere}
				open={openModal}
				onClose={handleModalState}
				onSubmit={handleSubmit}
			/>}`

Det er forklaringen hvertfall også kan man vurdere hva som er best løsning

Reply:
Jeg gjør kanskje noe galt, men når jeg prøver med `if(!isResolved(hentVeilederePromise)) return <></>` og å droppe den `useEffect`-en i modalen, så får jeg ikke vist veilederene på første åpning. 

Den andre versjonen tror jeg ikke vil funke fordi deltakere ikke har en veileder før de er tildelt en.

Litt kjedelig å gi opp her, da jeg har gjerne skulle droppet den effecten i modalen, men jeg får se på det mer senere tror jeg.